### PR TITLE
fix: unpacking of quaternion wasn't accounting for fixed point math

### PR DESCRIPTION
**Pull request by rpadrela**
test: re-enabled quaternion serialization test

### CODE REVIEW

File: src/lib/QuaternionUnPack.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public static Quaternion UnPack(QuaternionPackCommon.Info info)
             var c = info.c / QuaternionPackCommon.FLOAT_PRECISION_MULT;
             var maxIndex = info.maxIndex;
 
-            var d = (int)Math.Sqrt(1f - (a * a + b * b + c * c));
+            var d = (int)Math.Sqrt(Utility.FixedPointFactor * Utility.FixedPointFactor - (a * a + b * b + c * c));
```
Comment (eriksvedang) [inline src/lib/QuaternionUnPack.cs:40]: What is the FixedPointFactor?

**Review by eriksvedang — APPROVED** [review]
Looks good – I had one question out of curiosity though :)

File: src/lib/QuaternionUnPack.cs (hunk 1 above)
Comment (rpadrela) [reply src/lib/QuaternionUnPack.cs:40]: The factor that we use to convert from float to int and vice-versa. See FixedPointToFloat() in basal-dotnet for more details.

Comment (eriksvedang) [reply src/lib/QuaternionUnPack.cs:40]: Ah, I see. Thanks!

**Pull request merged**